### ISSUE TITLE
2.2.1: OCSP Stapling failed while using LibreSSL

### ISSUE THREAD

**Issue by csmk** [h2o/h2o#1274]
Hi,

After upgrading to 2.2.1, I got this:
```
[OCSP Stapling] testing for certificate file:/foo/bar/something.crt
fetch-ocsp-response (using LibreSSL 2.5.3)
sending OCSP request to http://ocsp.int-x3.letsencrypt.org/
Error querying OCSP responder
139730553144984:error:27FFF072:OCSP routines:CRYPTO_internal:server response error:/tmp/ebuild/build/work/portage/dev-libs/libressl-2.5.3/work/libressl-2.5.3/crypto/ocsp/ocsp_ht.c:251:Code=400,Reason=Bad Request
OpenSSL exitted abnormally: openssl ocsp -issuer /tmp/l9eJoQDnRz/issuer.crt -cert /foo/bar/something.crt -url http://ocsp.int-x3.letsencrypt.org/ -header Host=ocsp.int-x3.letsencrypt.org -noverify -respout /tmp/l9eJoQDnRz/resp.der :[/etc/h2o/ssl.conf:6] in command listen, [OCSP Stapling] temporary failed for file:/foo/bar/something.crt
```

This seems to be a regression of #1270, so the following change fixed the issue for me:

```diff
--- fetch-ocsp-response
+++ fetch-ocsp-response.new
@@ -88,7 +88,7 @@
 print STDERR "sending OCSP request to $ocsp_uri\n";
 my $resp = run_openssl(
     "ocsp -issuer $issuer_fn -cert $cert_fn -url $ocsp_uri"
-    . ($openssl_version =~ /^(OpenSSL 1\.|LibreSSL )/is ? " -header Host@{[$openssl_version =~ /^OpenSSL 1\.0\./is ? ' ' : '=']}$ocsp_host" : "")
+    . ($openssl_version =~ /^(OpenSSL 1\.|LibreSSL )/is ? " -header Host@{[$openssl_version =~ /^(OpenSSL 1\.0\.|LibreSSL )/is ? ' ' : '=']}$ocsp_host" : "")
     . " -noverify -respout $tempdir/resp.der " . join(' ', @ARGV),
     1,
 );
```

**Comment by kazuho** [comment]
Ah my bad. Thank you for the fix!

**Comment by kazuho** [comment]
Released 2.2.2 with your fix. Thank you very much.
